x/nft/spec/02_messages.md

```diff
@@ -0,0 +1,122 @@
+> Copyright (c) 2016-2021 Shanghai Bianjie AI Technology Inc. (licensed under the Apache License, Version 2.0)
+> Modifications Copyright (c) 2021, CRO Protocol Labs ("Crypto.org") (licensed under the Apache License, Version 2.0)
+
+# Messages
+
+## MsgIssueDenom
+
+This message defines a type of nft, there can be multiple nft of the same type
+
+| **Field** | **Type** | **Description**                                                                                                                  |
+| :-------- | :------- | :------------------------------------------------------------------------------------------------------------------------------- |
+| Name      | `Id`     | The denomination ID of the NFT, necessary as multiple denominations are able to be represented on each chain.                    |
+| Name      | `string` | The denomination name of the NFT, necessary as multiple denominations are able to be represented on each chain.                  |
+| Sender    | `string` | The account address of the user sending the NFT. By default it is __not__ required that the sender is also the owner of the NFT. |
+| Schema    | `string` | NFT specifications defined under this category                                                                                   |
+
+```go
+type MsgIssueDenom struct {
+    Id     string
+    Name   string
+    Schema string
+    Sender string
+}
+```
+
+## MsgTransferNFT
+
+This is the most commonly expected MsgType to be supported across chains. While each application specific blockchain will have very different adoption of the `MsgMintNFT`, `MsgBurnNFT` and `MsgEditNFT` it should be expected that most chains support the ability to transfer ownership of the NFT asset. The exception to this would be non-transferable NFTs that might be attached to reputation or some asset which should not be transferable. It still makes sense for this to be represented as an NFT because there are common queriers which will remain relevant to the NFT type even if non-transferable. This Message will fail if the NFT does not exist. By default it will not fail if the transfer is executed by someone beside the owner. **It is highly recommended that a custom handler is made to restrict use of this Message type to prevent unintended use.**
```

Comment:
> By default it will not fail if the transfer is executed by someone beside the owner. **It is highly recommended that a custom handler is made to restrict use of this Message type to prevent unintended use.**

@devashishdxt @yihuang @leejw51crypto

Reply:
@tomtau Is it possible to create issues for these things and do them in separate PRs. If we include all the changes in one PR, it'll become too big (which it already is) and it'll be difficult to review too.

We can keep the main issue open and add sub-issues to it.

Reply:
Somehow, in the code, it do check the sender is owner though.